### PR TITLE
Edit readme

### PR DESCRIPTION
**Pull request by 5im1z**
updated links, added taiga & gsheet section, changed background of figs for readability in dark mode

### CODE REVIEW

**Review by javadnoorb — APPROVED** [review]
thanks

File: README.md [hunk 1]
```diff
@@ -1,34 +1,27 @@
-# ccle_processing
+# depmap_omics
 
 __have a look at [DepMap](https://www.depmap.org)__
 
-![](https://github.com/broadinstitute/ccle_processing/blob/master/documentation/depmap-logo.png)
+![](https://github.com/broadinstitute/ccle_processing/blob/editREADME/documentation/depmap-logo_white.png)
```
Comment (jkobject) [inline README.md:5]: use relative paths.

File: README.md [hunk 2]
```diff
@@ -43,61 +36,70 @@ If you are not familiar with these notions, we will first recommend you get more
 - [Terra and gcp](https://docs.google.com/document/d/1zTtaN-Px64f8JvgydZNdBbzBpFWyZzEpshSNxQh43Oc/edit#heading=h.dz5wh0l4bu9g)
 
 
-### install it 
+### Installatiion <a name="installation"></a>
+
+`git clone http://github.com/BroadInstitute/depmap_omics.git && cd depmap_omics`
+
+`pip install .`
```
Comment (jkobject) [inline README.md:43]:
```suggestion
`pip install -e .`
```

File: README.md [hunk 3]
```diff
@@ -43,61 +36,70 @@ If you are not familiar with these notions, we will first recommend you get more
...
+## For Internal Users <a name="internal-users"></a>
 
-> Mind that you need to know about the tools listed above
-- Ask dbgap access to your supervisor
+> To learn about the tools we use in the pipeline, see [here](#running-pipeline) for a detailed walkthrough
 
 ### Getting Terra Access
 
 1. You will need to request access to the following terra workspaces:
   - https://app.terra.bio/#workspaces/broad-firecloud-ccle/DepMap_Mutation_Calling_CGA_pipeline
```
Comment (jkobject) [inline README.md:68]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -43,61 +36,70 @@ If you are not familiar with these notions, we will first recommend you get more
...
   - https://app.terra.bio/#workspaces/broad-firecloud-ccle/DepMap_hg38_RNAseq
   - https://app.terra.bio/#workspaces/broad-firecloud-ccle/DepMap_WES_CN_hg38
+  - https://app.terra.bio/#workspaces/broad-firecloud-ccle/DepMap_WGS_CN
```
Comment (jkobject) [inline README.md:71]: maybe explain it contains the mutation pipeline

File: README.md [hunk 5]
```diff
@@ -43,61 +36,70 @@ If you are not familiar with these notions, we will first recommend you get more
...
   - CCLE-PIPELINE
-5. If you need to get access to the depmap data:
-  - __Exome__ [IBM](https://firecloud.terra.bio/#workspaces/broad-genomics-delivery/Getz_IBM_CellLines_Exomes/data)
-  - __Exome__ [Broad](https://firecloud.terra.bio/#workspaces/broad-firecloud-ccle/CCLE_DepMap_WES)
-  - __Exome__ [Broad](https://firecloud.terra.bio/#workspaces/broad-genomics-delivery/CCLE_DepMap_WES)
-  - __rna__ [IBM](https://firecloud.terra.bio/#workspaces/broad-genomics-delivery/Getz_IBM_CellLines_RNASeqData/data)
-  - __rna__ [Broad](https://firecloud.terra.bio/#workspaces/broad-firecloud-ccle/CCLE_DepMap_RNAseq)
-  - __rna__ [Broad](https://firecloud.terra.bio/#workspaces/broad-genomics-delivery/CCLE_DepMap_RNAseq)
-6. Request access to the data bucket `gs://ccle_bams/`
+5. If you need to get access to the depmap data, use the following links:
```
Comment (jkobject) [inline README.md:82]: there is not the depmap data here. this is where the data is sent to us from GP. it gets erased at some point. I think Exome might soon not contain anything anymore..

File: README.md [hunk 6]
```diff
@@ -119,13 +121,13 @@ This involves comparing to the list in the Google sheet "Cell Line Profiling Sta
...
-> Mind that you need to know about the tools listed above
+> To learn about the tools we use in the pipeline, see [here](#running-pipeline) for a detailed walkthrough
 
 ### Creating your Terra Workspaces:
 
-1. Your first need a terra account with correct billing setup
+1. You first need a terra account with correct billing setup
 2. use the different *_pipeline folders, they contain everything to setup the workspace correctly
   1. use the WDL scripts available in:
     - https://dockstore.org/my-workflows/github.com/broadinstitute/depmap_omics/WGS_pipeline
```
Comment (jkobject) [inline README.md:133]: I think we would want more information for external users at some point... The goal of this github page is really for external users mostly

File: README.md [hunk 7]
```diff
@@ -153,27 +155,94 @@ The notebook architectures are as follows:
...
+  - [https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
+- __rsem__: 
+  - [https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323)
+- __star fusion__: 
+  - [https://github.com/STAR-Fusion/STAR-Fusion/wiki](https://github.com/STAR-Fusion/STAR-Fusion/wiki)
+  - [http://biorxiv.org/content/early/2017/03/24/120295](http://biorxiv.org/content/early/2017/03/24/120295)
+- __mutect__: 
+  - [https://software.broadinstitute.org/cancer/cga/mutect](https://software.broadinstitute.org/cancer/cga/mutect)
+  - [https://youtu.be/rN-cLrb5aGs](https://youtu.be/rN-cLrb5aGs)
+  - [https://gatk.broadinstitute.org/hc/en-us/articles/360036490432-Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360036490432-Mutect2)
```
Comment (jkobject) [inline README.md:178]: maybe mutect2 also?

File: README.md [hunk 8]
```diff
@@ -153,27 +155,94 @@ The notebook architectures are as follows:
...
 - in the initialization you might want to remove any import related to `taiga` and `gsheet` to not cause any errors.
 - feel free to reuse `createDatasetWithNewCellLines`, `GetNewCellLinesFromWorkspaces` or any other function for your own needs.
 
-### 2. Running Terra Pipelines
+### 2. Running Terra Pipelines <a name="running-terra-pipelines"></a>
+
+We are using Dalmatian to send requests to Terra, so before running this part, you need to make sure that your dalmatian `WorkspaceManager` object is initialized with the right workspace you created and that the functions take as input you workflow names. You also need to make sure that you created your sample set with all your samples and that you initialized the `sampleset` string with its name.
+You can then run the following two pipelines on your samples. The whole process may take 3-7 days.
+
+#### Copy Numbers
```
Comment (jkobject) [inline README.md:206]: we should not call it this way. it does CN and somatic mutations (and actually germline mutations and SV)

File: README.md [hunk 9]
```diff
@@ -153,27 +155,94 @@ The notebook architectures are as follows:
...
+
+We are running the following workflows in this order to generate copy number and mutation datasets:
+
+[WGS_pipeline](https://dockstore.org/workflows/github.com/broadinstitute/depmap_omics/WGS_pipeline:master?tab=info) imports and runs several sub-processes to generate copy number segments and MAF data.
+
+[WGS_aggregate](https://dockstore.org/workflows/github.com/broadinstitute/depmap_omics/WGS_aggregate:master?tab=info) aggregates CN segments and MAFs into their respective files.
+
+The output files for download will be saved under the sample set under the `combined_seg_file`, `filtered_CGA_MAF_aggregated` attribute.
```
Comment (jkobject) [inline README.md:214]: list them the same way as below?

Comment (jkobject) [reply README.md:214]: we also use unfiltered CGA maf

and we use some more files for QC

File: README.md [hunk 10]
```diff
@@ -153,27 +155,94 @@ The notebook architectures are as follows:
...
+*   rsem_genes_tpm
+*   rsem_transcripts_tpm
+*   fusions_star
+
+### On local
+
+__We then save the workflow configurations used__
+__,delete unmapped bams generated during the process__
+__and move the hg38 aligned bams to our own datastorage bucket__
+__We download and reprocess removing the appended version and keeping only the newest versions__
```
Comment (jkobject) [inline README.md:237]: I don't understand that last line

File: README.md [hunk 11]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
 
 _You would only be interested here at minima in the result downloading_
  
-...and post processing tasks.
+...and postprocessing tasks. The main postprocessing steps for each pipeline are as followed:
+
+#### CN
 
-> Unfortunately for now the postProcessing tasks are not all made to be easily run outside of the CCLE pipeline. Most of them are in R and are run with the Rpy2 tool.
+`copynumbers.py` contains the main postprocessing function (`postProcess()` and their wrappers for internal use) responsible for postprocessing segments and gene-level CN files.
```
Comment (jkobject) [inline README.md:261]:
```suggestion
`copynumbers.py` contains the main postprocessing function (`postProcess()` and their wrappers for internal use) responsible for postprocessing segments and creating gene-level CN files.
```

File: README.md [hunk 12]
```diff
@@ -277,235 +372,7 @@ The data we are presenting comes from different WES targets/baits/intervals.
...
-
-https://docs.google.com/document/d/1O6ZWGAnG6CO_smtaA-fqQmSqV64TF1GHFbj-ZeIqD6U/edit?usp=sharing
-
-https://docs.google.com/document/d/1qHnqO3QGdubu8IvB9zI13Z48EbKwDTpgRg0TnT6GULw/edit?usp=sharing
-
-https://docs.google.com/document/d/189K81MOIYlvg4ePvtHPBSfMen1sgYm7G1EncG5N8tFk/edit?usp=sharing
 
-https://docs.google.com/document/d/19MvCIpRID12vfIlc5i2XmJB8yT2jJ2xwNM_PI1HRC0A/edit?usp=sharing
 
 @[jkobject](https://www.jkobject.com)
```
Comment (jkobject) [inline README.md:377]: feel free to add your name too now :)

File: README.md [hunk 13]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
...
+
+#### CN
 
-> Unfortunately for now the postProcessing tasks are not all made to be easily run outside of the CCLE pipeline. Most of them are in R and are run with the Rpy2 tool.
+`copynumbers.py` contains the main postprocessing function (`postProcess()` and their wrappers for internal use) responsible for postprocessing segments and gene-level CN files.
 
-So amongst these functions, some of them might be of a lesser interest to an external user. The most important ones for each pipelines are:
+#### Mutation
+
+`postProcess()` in `mutations.py` is responsible for postprocessing aggregated MAF files. Here, rather than rerunning the entire analysis, because we know we are adding only WES samples, we can download the previous release's MAF, add the samples, update any annotations, and perform any global filters at the end.
```
Comment (jkobject) [inline README.md:265]: this is not what we do actually we rerun everything everytime for the postprocessing. It is actually bad practice for a pipeline to directly use results from run n-1.

File: README.md [hunk 14]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
...
+
+#### RNA
 
-- `processSegments`
-- `interpolateGapsInSegmented`
-- `extendEndsOfSegments`
-- `generateEntrezGenes`
-- `generateGeneLevelMatrixFromSegments`
+`expressions.py` contains the main postprocessing function (`postProcess()` and their wrappers for internal use) responsible for postprocessing aggregated expression data from RSEM, which removes duplicates, renames genes, filters and log transforms entires.
```
Comment (jkobject) [inline README.md:269]: also makes protein level expression

File: README.md [hunk 15]
```diff
@@ -277,235 +372,7 @@ The data we are presenting comes from different WES targets/baits/intervals.
 We are currently using Illumina ICE intervals and Agilent intervals. you can find their related PON files and interval files as parameters in our workspace files in `data/xQx.json`
```
Comment (jkobject) [inline README.md:373]:
```suggestion

__additional auxilliary data is used and listed in some of our workflow, like the CGA pipeline. You will be able to find them by looking at the wdl scripts of each pipelines and looking into the data/xQx.json` files for workspace data files.__
```

File: README.md [hunk 16]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
...
-- `prepare_depmap_\*\_for_taiga`
 
 **Remarks:**
 - in the RNAseq pipeline we have an additional sub-pipeline at the end of the notebook to process the fusion calls from starFusion
 - to get the exact same results as in CCLE, be sure to run `genecn = genecn.apply(lambda x: np.log2(1+x))` to the genecn dataframe in the CNV pipeline (present at the end of the validation steps).
 - we do not yet have integrated our germline calling in the mutation pipeline but you can still find the haplotypeCaller\|DeepVariant workflows and their parameters
 
 
-### 4. QC, grouping and uploading to the portal
+### 4. QC, Grouping and Uploading to the Portal <a name="qc-grouping-uploading"></a>
```
Comment (jkobject) [inline README.md:288]: We should say this is internal user specific..

File: README.md [hunk 17]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
...
+__Compare to previous release (broad only)__
 
-## File structure
+We compare the results to the previous releases MAF. Namely:
+
+- Count the total number of mutations per cell line, split by type (SNP, INS, DEL)
+- Count the total number of mutations observed by position (group by chromosome, start position, end position and count the number of mutations)
+- Look at specific differences between the two MAFs (join on DepMap_ID, Chromosome, Start position, End position, Variant_Type). This is done for WES only
+
+__check if important mutations are present or not__
```
Comment (jkobject) [inline README.md:311]:
```suggestion
```

File: README.md [hunk 18]
```diff
@@ -185,49 +254,75 @@ This step will do a set of tasks:
...
+
+#### RNA
+
+Once the expression files are saved, we do the following validations:
+- mean,max,var...
+- to previous version: same mean, max, var...
+- we QC on the amount of genes with 0 counts for each samples
+
+
+Besides QC, we are also preparing the data to be released to different groups, removing the samples per access category: Blacklist\|Internal\|DepMapConsortium\|Public.
```
Comment (jkobject) [inline README.md:322]:
```suggestion
> Besides QC, we are also preparing the data to be released to different groups, removing the samples per access category: Blacklist\|Internal\|DepMapConsortium\|Public.
```

**Review by jkobject — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (5im1z) [reply README.md:81]: should the WES links be removed now or should we wait?

File: README.md [hunk 19]
```diff
@@ -43,61 +36,69 @@ If you are not familiar with these notions, we will first recommend you get more
...
+- In order to run the imports [gsheets](https://pypi.org/project/gsheets/), you need to make sure your broad google account has access to the cell line info sheets. In addition, you will need to obtain the following google API credential files:
+  - Go to [console](https://console.developers.google.com/), acquire access if you don't have it already. Under credentials -> create credentials -> OAuth Client ID -> application type = Desktop app -> create. Download `client_secreat.json` and save as `~/.client_secret.json` (Refer to quickstart [here](https://gsheets.readthedocs.io/en/stable/)).
+  - After calling `gsheets.from_files` for the first time using `~/.client_secret.json`, log in via google following the prompt in browser, `storage.json` will be created automatically. Save it as `~/.storage.json`.
+  - Follow instruction [here](https://cloud.google.com/docs/authentication/production?authuser=1#create_service_account), create service account if needed, and save key file as `../.credentials.json`.
 
+*Remember to share relevant gsheets with the service account (`client_email` in `../.credentials.json`).
 
-### boot up
+### Boot up
```
Comment (5im1z) [inline README.md:101]: @jkobject is this outdated, considering creating virtual release folders is handled by DepMap_upload.ipynb now?

**Review by javadnoorb — APPROVED** [review]

**Pull request merged**